### PR TITLE
[이지훈] 1주차 정리

### PR DESCRIPTION
**Pull request by huncozyboy**
1. 영속성 컨텍스트의 엔티티 관리시 장점은 ?
2. flush 모드 준영속의 개념과 예시는 ?
3. 패러다임의 불일치를 해결하는 JPA의 두가지 방법은 ?

### CODE REVIEW

**Comment by hyxklee** [comment]
1. 1차 캐시를 이용한 조회 성능 극대화, 쓰기 지연을 통한 쿼리 성능 상승, 변경 감지로 인한 엔티티 상태 변경시 자동으로 DB에 반영
2. flush는 영속성 컨텍스트의 변경 내용을 DB에 반영하는 것을 뜻하며 이 시점에 SQL 저장소에 쌓인 쿼리 문이 DB에 반영됨 / 준영속은 영속성 컨텍스트에 저장되었다가 detach(), clear(), close()되어 준영속 컨텍스트에서 제거된 상태를 뜻함. 그렇게 때문에 영속성 컨텍스트의 지원을 받을 수 없음
3. 객체와 엔티티는 서로의 패러다임이 다르기 때문에 불일치 문제가 발생. 1. 상속 관계의 객체를 DB에 저장할 때 테이블을 나눠 저장해야하는데 JPA는 이를 자동으로 해결 2. 객체는 참조를 사용하고 테이블은 외래키를 이용하는데 이대로라면 객체에 테이블 구조에 맞는 외래 키들을 객체에 멤버 변수로 생성해 같이 관리해야함. JPA를 사용하면 자동으로 적절한 쿼리문을 생성해줌

**Comment by anxi01** [comment]
1. 1차 캐시, 동일성(==)보장, 트랜잭션 쓰기 지연, 더티체킹, 지연 로딩 등이 있습니다.
2. 준영속 : 영속상태였다가 영속성 컨텍스트에서 관리하지 않는 상태를 의미, 
   특정 엔티티 준영속 : em.detach(), 전체 준영속 : em.clear(), close()
3. 상속 (자바 컬렉션에 객체 저장하듯이 JPA에 저장), 연관관계 (테이블에서 사용하는 외래키가 아닌 객체 참조에 접근해서 조회)

**Comment by yechan-kim** [comment]
1. 1차 캐시, 동일성 보장, 트랜잭션을 지원하는 쓰기 지연, 변경 가지, 지연 로딩
2. 영속성 컨텍스트가 관리하는 영속 상테의 엔티티가 영속성 컨텍스트에서 분리된 상태
ex) em.detach(), em.clear()
3. 상속, 연관관계
